exercises/strain/example.scala

```diff
@@ -0,0 +1,7 @@
+object Strain {
+  def keep[A](traversable: Traversable[A], func: A => Boolean): Traversable[A]
+    = for (a <- traversable if func(a)) yield a
```

Comment:
One could argue that a for-comprehension is just syntactic sugar for `filter` and therefore violates the Readme restrictions?! :)